CONTRIBUTING.md

```diff
@@ -1,35 +1,99 @@
+By contributing to this project you agree to the Developer Certificate of Origin (DCO). 
+This document was created by the Linux Kernel community and is a simple statement that you, as a contributor, have the legal right to make the contribution. See the [DCO](DCO) file for details.
 
-## Getting Started
+# Notes
+Cluster Monitoring Operator is part of OpenShift and therefore follows the [OpenShift Life Cycle](https://access.redhat.com/support/policy/updates/openshift)
 
-- Fork the repository on GitHub
-- Read the [README](README.md) for build and test instructions
-- Play with the project, submit bugs, submit patches!
+You should keep this in mind when decding in which release you want your feature or fix.
 
-## Contribution Flow
+# Contribution Flow
+Before you get started, you have to perform the following **mandatory** steps:
+* Open an bug in Bugzilla
+* Fork this repository
 
-This is a rough outline of what a contributor's workflow looks like:
+If you want to make changes to the actual code, please follow the [Coding Style](#coding-style) for code changes.
 
-- Create a topic branch from where you want to base your work (usually master).
-- Make commits of logical units.
-- Make sure your commit messages are in the proper format (see below).
-- Push your changes to a topic branch in your fork of the repository.
-- Make sure the tests pass, and add any new tests as appropriate.
-- Submit a pull request to the original repository.
+## General workflow information
+These steps outline the general contribution workflow:
 
-Thanks for your contributions!
+* Create a topic branch from where you want to base your work (usually master).
+* Make commits of logical units.
+* Make sure your commit messages are in the proper format (see [Format of the Commit Message](#format-of-the-commit-message))
+* Push your changes to a topic branch in your fork of the repository.
+* Make sure the tests pass, and add any new tests as appropriate.
+* Submit a pull request to the original repository. (see [Format of Pull Requests](#format-of-pull-requests))
 
-### Coding Style
+## Working with jsonnet
+This project is making use of a lot of upstream projects and imports them.
+To work with jsonnet you should have [jsonnet bundler](https://github.com/jsonnet-bundler/jsonnet-bundler) installed and [updated](https://github.com/coreos/kube-prometheus#update-jb).
 
-cluster-monitoring-operator projects written in Go follow a set of style guidelines that we've documented [here](https://github.com/coreos/docs/tree/master/golang). Please follow them when working on your contributions.
+Assuming you have made your changes upstream ([see an example change](https://github.com/kubernetes-monitoring/kubernetes-mixin/pull/466/files)),
+you can now go ahead and update the dependency here:
 
-### Format of the Commit Message
+```
+cd jsonnet
+jb update
```

Comment:
Preferred way now is to execute  `make jsonnet/vendor --always-make`. This ensures correct version of `jb` is used.